language/stdlib/modules/Libra.move

```diff
@@ -804,6 +809,8 @@ module Libra {
     spec schema RegisterCurrencyAbortsIf<CoinType> {
         lr_account: signer;
         currency_code: vector<u8>;
+        scaling_factor: u64;
+        aborts_if scaling_factor == 0 with Errors::INVALID_ARGUMENT;
         aborts_if !Roles::spec_has_register_new_currency_privilege_addr(Signer::spec_address_of(lr_account))
```

Comment:
As you are here, could you introduce `Roles::assert_new_currency_privilege_addr` and the matching schema in Roles, following the existing patterns?

Reply:
I am in enough trouble as is.  Let us leave this for a separate PR.